backend/fms_core/serializers.py

```diff
@@ -667,21 +673,39 @@ class Meta:
             'pedigree',
             ]
 
+class StepSpecificationSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = StepSpecification
+        fields = ("id", "display_name", "sheet_name", "column_name", "value")
+
 class StepSerializer(serializers.ModelSerializer):
+    step_specifications = StepSpecificationSerializer(read_only=True, many=True)
     class Meta:
         model = Step
-        fields = ["id", "name", "protocol_id"]
+        fields = ["id", "name", "protocol_id", "step_specifications"]
 
-class WorkflowSerializer(serializers.ModelSerializer):
-    steps = serializers.SerializerMethodField()
+class StepWithOrderSerializer(serializers.ModelSerializer):
```

Comment:
Not sure i like the StepWithOrder serializer name and content. I think we should review this to be a StepOrder serializer that also include a step name and protocol_id for convenience but at the moment it has a unique step id but the order is not related uniquely to that id but instead to the step order instance. This could cause issues. I would use the StepOrderSerializer (step_order_id, step_id, step_name, protocol_id, worflow_id, order) for the workflow. It is not a big change i think but it might make it more usefull and less confusing.